crates/adapters/src/static_compile/serialize_with_context.rs

```diff
@@ -0,0 +1,489 @@
+        where
+            $($arg: $crate::SerializeWithContext<C>),*
+            $($($arg : $bound)?),*
+        {
+            fn serialize_with_context<S>(&self, serializer: S, context: &C) -> Result<S::Ok, S::Error>
+            where
+                S: serde::Serializer,
+            {
+                let mut struct_serializer = serializer.serialize_struct(stringify!($struct), $num_fields)?;
+                $(
+                    serde::ser::SerializeStruct::serialize_field(&mut struct_serializer, $column_name, &$crate::SerializationContext::new(context, &self.$field_name))?;
+                )*
+                serde::ser::SerializeStruct::end(struct_serializer)
+            }
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! serialize_table_record {
+    ($struct:ident[$num_fields:expr]{$($field_name:ident[$column_name:tt]: $type:ty),* }) => {
+        #[allow(unused_variables)]
+        #[allow(unused_mut)]
+        impl $crate::SerializeWithContext<$crate::SqlSerdeConfig> for $struct {
+            fn serialize_with_context<S>(&self, serializer: S, context: & $crate::SqlSerdeConfig) -> Result<S::Ok, S::Error>
+            where
+                S: serde::Serializer,
+            {
+                let mut struct_serializer = serializer.serialize_struct(stringify!($struct), $num_fields)?;
+                $(
+                    serde::ser::SerializeStruct::serialize_field(&mut struct_serializer, $column_name, &$crate::SerializationContext::new(context, &self.$field_name))?;
+                )*
+                serde::ser::SerializeStruct::end(struct_serializer)
+            }
+        }
+    }
+}
+
+serialize_struct!(NeighborhoodDescr(K, V)[4]{
+    anchor["anchor"]: Option<K>,
+    anchor_val["anchor_val"]: V,
+    before["before"]: usize,
+    after["after"]: usize
+});
+
+#[cfg(test)]
+mod test {
+
+    use dbsp::algebra::F64;
+    use lazy_static::lazy_static;
+    use rust_decimal::Decimal;
+    use rust_decimal_macros::dec;
+
+    use crate::{SerializationContext, SerializeWithContext, SqlSerdeConfig};
+
+    #[derive(Debug, Eq, PartialEq)]
+    struct TUPLE0;
+    serialize_struct!(TUPLE0()[0] {});
+
+    lazy_static! {
+        static ref DEFAULT_CONFIG: SqlSerdeConfig = SqlSerdeConfig::default();
+    }
+
+    fn serialize_json_with_default_context<T>(val: &T) -> Result<String, serde_json::Error>
+    where
+        T: SerializeWithContext<SqlSerdeConfig>,
+    {
+        let mut result: Vec<u8> = Vec::new();
+        val.serialize_with_context(
+            &mut serde_json::Serializer::new(&mut result),
+            &DEFAULT_CONFIG,
+        )?;
+        Ok(String::from_utf8(result).unwrap())
+    }
+
+    #[test]
+    fn serialize_unit() {
+        assert_eq!(serialize_json_with_default_context(&TUPLE0).unwrap(), "{}");
+    }
+
+    #[derive(Debug, Eq, PartialEq)]
+    #[allow(non_snake_case)]
+    struct Struct2 {
+        #[allow(non_snake_case)]
```

Comment:
why is this `allow` needed?

Reply:
maybe you can also test a float. At least in SQL floats require an E to be distinguished from decimals.
e.g., 1.0E0.

Reply:
Just added a float test to the end-to-end snowflake test to make sure they make it all the way through.